data.sql

```diff
@@ -228,3 +228,14 @@ INSERT INTO visits(animal_id, vet_id, date_of_visit) VALUES
   '20210111'
 );
 COMMIT;
+
+-- This will add 3.594.280 visits considering you have 10 animals, 4 vets, and it will use around ~87.000 timestamps (~4min approx.)
+INSERT INTO visits (animals_id, vets_id, date_of_visit) 
+SELECT * FROM (SELECT id FROM animals) animal_ids, 
+(SELECT id FROM vets) vets_ids, 
+generate_series('1980-01-01'::timestamp, '2021-01-01', '4 hours') 
+visit_timestamp;
+
+-- This will add 2.500.000 owners with full_name = 'Owner <X>' and email = 'owner_<X>@email.com' (~2min approx.)
+Insert into owners (full_name, email) select 'Owner ' || generate_series(1,2500000),
+ 'owner_' || generate_series(1,2500000) || '@mail.com';
```

Comment:
- [ ] Great work writing these `INSERT` statements. After populating the necessary tables with these data, the outcome of each of the following queries written below should have an execution time that is **_greater or equal_** to `1000ms`. 

- ` EXPLAIN ANALYZE SELECT COUNT(*) FROM visits where animal_id = 4;`
- `EXPLAIN ANALYZE SELECT * FROM visits where vet_id = 2;`
- `EXPLAIN ANALYZE SELECT * FROM owners where email = 'owner_18327@mail.com';`

I would advise you to run the statements that insert data into the table two or three more times, make the `explain analyze` statement and take screenshots of the execution time that are equal to or more than _**1 second (1000ms)**_. 👍 😎 

Kindly attach the screenshots of the queries to your pull request review showing with each screenshot display an execution time of _**>=1000**_. After these, you then rerun the SQL statements that apply indexes to the special columns and then run each of the queries above again. 😎 

Thanks to the indexes applied, their execution time should be **_significantly lower_** than `1000ms`. Take the screenshots and attach them to your pull request summary. I understand that the current screenshots after running the indexes are significantly low, but you would agree that if you insert more data into the tables, the execution time will be different even though you applied the indexes.

Kindly implement these changes. 👍 😎 

#### Hint:

- Run the queries before applying the indexes. Take screenshots of **_execution time above 1000ms_**
- Apply the indexes
- Run the queries again. Take screenshots of execution time **_significantly below 1000ms_**.

Reply:
## Changes ♻️
- [x] The statements that insert data into the table executed three more times
- [x] Execution time above 1000ms
![explain1b](https://github.com/Benawi/Vet-clinic-database/assets/21217148/43cafa28-aa93-4adb-b57e-68814bc8ddf6)
![explain2b](https://github.com/Benawi/Vet-clinic-database/assets/21217148/c8c3878b-b9f3-4dc2-9a29-affeb0079efa)
![explain3b](https://github.com/Benawi/Vet-clinic-database/assets/21217148/40608e1f-ec3a-46ac-ab16-5c33b749cc49)
- [x] Execution time significantly below 1000ms.
![explain1ba](https://github.com/Benawi/Vet-clinic-database/assets/21217148/176c9ef4-4ffd-4616-a3c4-361b9f597185)
![explain2ba](https://github.com/Benawi/Vet-clinic-database/assets/21217148/7e993af8-84ea-4773-8796-32def9fdd63f)
![explain3ba](https://github.com/Benawi/Vet-clinic-database/assets/21217148/40a43303-0b47-4282-a3ed-308c82456f65)